test/signer.ts

```diff
@@ -0,0 +1,886 @@
+        });
+
+        it('should sign expiration', async () => {
+          await signer['getSignedUrlV2'](CONFIG);
+
+          const blobToSign = authClientSign.getCall(0).args[0];
+          assert(blobToSign.includes(CONFIG.expiration.toString(10)));
+        });
+
+        it('should sign canonical headers', async () => {
+          sandbox
+            .stub(signer, 'getCanonicalHeaders')
+            .returns('canonical-headers');
+          await signer['getSignedUrlV2'](CONFIG);
+
+          const blobToSign = authClientSign.getCall(0).args[0];
+          assert(blobToSign.includes('canonical-headers'));
+        });
+
+        it('should sign resource path', async () => {
+          sandbox.stub(signer, 'getResourcePath').returns('/resource/path');
+          await signer['getSignedUrlV2'](CONFIG);
+
+          const blobToSign = authClientSign.getCall(0).args[0];
+          assert(blobToSign.endsWith('/resource/path'));
+        });
+
+        it('should compose blobToSign without contentMd5 and contentType', async () => {
+          sandbox
+            .stub(signer, 'getCanonicalHeaders')
+            .returns('canonical-headers');
+          sandbox.stub(signer, 'getResourcePath').returns('/resource/path');
+          await signer['getSignedUrlV2'](CONFIG);
+
+          const blobToSign = authClientSign.getCall(0).args[0];
+          assert.strictEqual(
+            blobToSign,
+            [
+              'GET',
+              '',
+              '',
+              CONFIG.expiration,
+              'canonical-headers' + '/resource/path',
+            ].join('\n')
+          );
+        });
+      });
+
+      it('should return v2 query', async () => {
+        const query = (await signer['getSignedUrlV2'](CONFIG)) as Query;
+        assert.deepStrictEqual(query, {
+          GoogleAccessId: CLIENT_EMAIL,
+          Expires: CONFIG.expiration,
+          Signature: 'signature',
+        });
+      });
+
+      it('rejects with SigningError on signing Error', () => {
+        const err = new Error('my-err');
+        err.stack = 'some-stack-trace';
+        sandbox.stub(authClient, 'sign').rejects(err);
+
+        assert.rejects(() => signer['getSignedUrlV2'](CONFIG), {
+          name: 'SigningError',
+          message: 'my-err',
+          stack: 'some-stack-trace',
+        });
+      });
+    });
+
+    describe('getSignedUrlV4', () => {
+      let signer: URLSigner;
+      let CONFIG: GetSignedUrlConfigInternal;
+
+      beforeEach(() => {
+        signer = new URLSigner(authClient, bucket, file);
+        CONFIG = {
+          method: 'GET',
+          expiration: Math.floor((NOW.valueOf() + 2000) / 1000),
+          bucket: bucket.name,
+        };
+      });
+
+      it('should fail for expirations beyond 7 days', () => {
+        CONFIG.expiration = NOW.valueOf() + 7.1 * 24 * 60 * 60;
+
+        assert.throws(() => {
+          signer['getSignedUrlV4'](CONFIG);
+        }, /Max allowed expiration is seven days/);
```

Comment:
Can you check the full error message, `Max allowed expiration is seven days (${SEVEN_DAYS} seconds).`?